mirascope/core/bedrock/_utils/_extract_stream.py

```diff
@@ -0,0 +1,50 @@
+from collections.abc import AsyncGenerator, Callable, Coroutine, Generator
+from functools import wraps
+from typing import Any, ParamSpec
+
+from mypy_boto3_bedrock_runtime.type_defs import (
+    ConverseStreamResponseTypeDef,
+)
+from types_aiobotocore_bedrock_runtime.type_defs import (
+    ConverseStreamResponseTypeDef as AsyncConverseStreamResponseTypeDef,
+)
+
+from mirascope.core.bedrock._utils._types import (
+    AsyncStreamOutputChunk,
+    StreamOutputChunk,
+)
+
+_P = ParamSpec("_P")
+
+
+def _extract_sync_stream_fn(
```

Comment:
nit: these functions should not be private, it's the module that is private, and these are public in the private module (see other utilities for reference)